File: src/main/java/ru/citeck/idea/codeinsight/forms/components/InputComponent.java
package ru.citeck.idea.codeinsight.forms.components;

import ecos.com.fasterxml.jackson210.annotation.JsonIgnore;
import lombok.Getter;
import lombok.Setter;
import ru.citeck.idea.codeinsight.forms.Properties;

import java.util.Set;

@Getter
public abstract class InputComponent extends Component {

    private final Boolean optionalWhenDisabled = false;
    private final Boolean clearOnHide = false;
    private final Properties properties = new Properties();
    @Setter
    private Boolean disableInlineEdit = true;

    @Override
    boolean getInput() {
        return true;
    }

    @JsonIgnore
    public abstract Set<String> getSupportedArtifactTypes();

}

File: src/main/kotlin/ru/citeck/idea/artifacts/action/codeinsight/forms/components/InputComponent.kt
package ru.citeck.idea.artifacts.action.codeinsight.forms.components

import ecos.com.fasterxml.jackson210.annotation.JsonIgnore
import ru.citeck.idea.artifacts.action.codeinsight.forms.Properties

abstract class InputComponent : ru.citeck.idea.artifacts.action.codeinsight.forms.components.Component() {

    val optionalWhenDisabled = false
    val clearOnHide = false
    val properties = ru.citeck.idea.artifacts.action.codeinsight.forms.Properties()

    var disableInlineEdit = true

    override val input: Boolean
        get() = true

    @JsonIgnore
    abstract fun getSupportedArtifactTypes(): Set<String>
}
